File: Codesignal/Eric/68.AllLongestStrings.js
function solution(input) {
    sorted = [...input]
    sorted.sort((a, b) => a.length - b.length);
    const length = sorted[sorted.length - 1].length;
    return sorted.filter((item) => item.length === length);
}

console.log(solution(["a", 
"abc", 
"cbd", 
"zzzzzz", 
"a", 
"abcdef", 
"asasa", 
"aaaaaa"]))

File: Codesignal/Eric/68.AllLongestStrings.py
def solution(input):
    length = len(sorted(input, key=len)[-1])
    return [x for x in input if len(x) == length]
    
    
print(solution(["a", 
 "abc", 
 "cbd", 
 "zzzzzz", 
 "a", 
 "abcdef", 
 "asasa", 
 "aaaaaa"]))

